### DISCUSSION TITLE
Attention weights in TransformerConv

### DISCUSSION THREAD

**Discussion by VaniOFX** [pyg-team/pytorch_geometric#2803] [Q&A]
Hi Matt, I was wondering if there is a reason why `TransformerConv` does not have a `return_attention_weights` just like `GATConv`? What do you think would be the easiest way to get the attention scores for the edges?

**Comment by wsad1** [comment]
I guess one way would be to implement a `return_attention_weights` functionality. Another simple hack would be to add this line `self.alpha=alpha` to the  `message` function in `TransformerConv`, and get `alpha` from the object of `TransformerConv` after calling `forward`.

**Reply by rusty1s** [reply]
Yes, I think we should definitely provide such functionality. Putting it on the roadmap.

**Reply by rusty1s** [reply]
Fixed in https://github.com/rusty1s/pytorch_geometric/pull/2807 thanks to @wsad1
